gcc/rust/checks/errors/rust-unsafe-checker.cc

```diff
@@ -262,10 +262,23 @@ UnsafeChecker::visit (MethodCallExpr &expr)
 void
 UnsafeChecker::visit (FieldAccessExpr &expr)
 {
-  // FIXME: If the receiver is an union, we need to be in an unsafe context to
-  // access it. Make sure to check.
-
   expr.get_receiver_expr ()->accept_vis (*this);
+
+  if (is_unsafe_context ())
+    return;
+
+  TyTy::BaseType *receiver_ty;
+  context.lookup_type (expr.get_receiver_expr ()->get_mappings ().get_hirid (),
```

Comment:
This should be:

```
bool ok = context.lookup_type();
rust_assert(ok);
```

I need to make a new interface for code generation, which can default to an error node, but there is some refactoring required for that.

Reply:
Fixed in [f532ae5](https://github.com/Rust-GCC/gccrs/pull/1416/commits/f532ae5aa37c8555dd6a99d8502436509db7930f)